addons/nightvision/XEH_preInit.sqf

```diff
@@ -2,6 +2,23 @@
 
 ADDON = false;
 
+PREP_RECOMPILE_START;
 #include "XEH_PREP.hpp"
+PREP_RECOMPILE_END;
+
+// Credit to killzonekid
+// Almost the same as (([] call CBA_fnc_getFov) select 1) but 10x faster
+DFUNC(kkTrueZoom) = {
+    (
+    [0.5,0.5]
+    distance2D
+    worldToScreen
+    positionCameraToWorld
+    [0,3,4]
+    ) * (
+    getResolution
+    select 5
+    ) / 2
```

Comment:
I don't trust the `distance` and `distance2D` commands. They're meant for terrains and not for vector math. If the terrain is sloped at the positions you are checking, it's like you are measuring the distance on a bend coordinate system, due to height differences. The command also strangely reports different values for the same position on different machines. I think some graphic settings responsible for terrain rendering affect the result too.